src/codegen.cpp

```diff
@@ -4014,6 +4037,8 @@ emit_gcpops(jl_codectx_t *ctx)
 {
     Function *F = ctx->f;
     for(Function::iterator I = F->begin(), E = F->end(); I != E; ++I) {
+        if (I->getTerminator() == NULL)
```

Comment:
You'd rather spell it as `assert(I->getTerminator() != NULL)`? Sure.

I added it before discovering that skipping over expressions was creating unterminated BasicBlocks, thus causing segfaults when code generation hit this method.